File: vue/src/api/role.js
import request from '@/utils/request'

export function getRoles() {
  return request({
    url: '/roles/get',
    method: 'get'
  })
}

export function addRole(data) {
  return request({
    url: '/rolegroup/add',
    method: 'post',
    data
  })
}


export function getRoleGroupPerm(id) {
  return request({
    url: `/rolegroup/perm/get?id=${id}`,
    method: 'get'
  })
}

export function editRole(data) {
  return request({
    url: '/rolegroup/edit',
    method: 'post',
    data
  })
}

export function getPermTemplate(data) {
  return request({
    url: '/rolegroup/template',
    method: 'post',
    data
  })
}

export function removeRole(id) {
  return request({
    url: '/rolegroup/remove',
    method: 'get',
    params: { id }
  })
}

export function roleList() {
  return request({
    url: `/rolegroup/list`,
    method: 'post'
  })
}

export function roleGroupName() {
  return request({
    url: `/role/list`,
    method: 'post'
  })
}

File: go/bug/route/role.go
package route

import (
	"itflow/bug/handle"

	"github.com/hyahm/xmux"
)

var Role *xmux.GroupRoute

func init() {
	Role = xmux.NewGroupRoute("role")
	Role.Pattern("/role/add").Post(handle.AddRole)
	Role.Pattern("/role/edit").Post(handle.EditRole)
	Role.Pattern("/role/list").Post(handle.RoleList)
	Role.Pattern("/role/remove").Get(handle.RoleDel)
	Role.Pattern("/role/get").Get(handle.GetRoles)
	Role.Pattern("/role/groupname").Post(handle.RoleGroupName)
}
